Shared/Repository/GenericRepository.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  GenericRepository.swift
+//  InternxtDesktop
+//
+//  Created by Patricio Tovar on 23/7/24.
+//
+
+import Foundation
+
+protocol GenericRepositoryProtocol {
```

Comment:
A repository should contain generic functions, that works for retrieving and updating the data, some examples of functions that a repository could contain:

`findById`
`updateById`
`deleteById`

Working with entities such `SyncedNodes` here is not a good idea, cause we are breaking the abstraction